cmd/kube-controller-manager/app/core.go

```diff
@@ -331,6 +332,10 @@ func startNamespaceController(ctx ControllerContext) (http.Handler, bool, error)
 	nsKubeconfig.QPS *= 20
 	nsKubeconfig.Burst *= 100
 	namespaceKubeClient := clientset.NewForConfigOrDie(nsKubeconfig)
+	return startModifiedNamespaceController(ctx, namespaceKubeClient, nsKubeconfig)
+}
+
+func startModifiedNamespaceController(ctx ControllerContext, namespaceKubeClient clientset.Interface, nsKubeconfig *restclient.Config) (http.Handler, bool, error) {
```

Comment:
This name isn't obvious to me, but if Walter liked it I won't nit it to death. Next time something more obvious?   If you removed a context dependency for instance it may make it more obvious.  Based on where you split it, I'm guessing you didn't want to deal with the client built from config because you can't mock that.